server/src/middleware/authorization.ts

```diff
@@ -0,0 +1,22 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import { NextFunction, Request, Response } from 'express';
+import passport from 'passport';
+
+import { JwtClaims } from '../controllers/auth';
+import { HttpCode } from '../types/httpCode';
+
+export function authorization(req: Request, res: Response, next: NextFunction): void {
```

Comment:
Shouldn't this be `authentication`? Or is this intended to also check for authorization in the future?

Reply:
Yes, I implemented this with authorization (role checking from token) in mind. We can extend this to handle the "system level" role check, if we implement it as planned in #228.